### PR TITLE
Change MetricDataPointFlags to the new immutable pattern

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan <bogdandrutu@gmail.com>

### CODE REVIEW

**Comment by dmitryax** [comment]
Can you PTAL at failing contrib tests. They look related

**Comment by bogdandrutu** [comment]
@dmitryax they are caused by https://github.com/open-telemetry/opentelemetry-collector/pull/5996, and will be fixed by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13738

File: pdata/plog/log_record_flags.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ var DefaultLogRecordFlags = LogRecordFlags(0)
 // defined in W3C Trace Context specification. 24 most significant bits are reserved and must be set to 0.
 type LogRecordFlags uint32
 
-// IsSampled returns true if the LogRecordFlags contains the TraceFlags.IsSampled flag.
+// IsSampled returns true if the LogRecordFlags contains the TraceFlags.NoRecordedValue flag.
```
Comment (TylerHelmuth) [inline pdata/plog/log_record_flags.go:25]:
```suggestion
// IsSampled returns true if the LogRecordFlags contains the TraceFlags.IsSampled flag.
```

**Review by TylerHelmuth — APPROVED** [review]

File: pdata/plog/log_record_flags.go (hunk 1 above)
Comment (bogdandrutu) [reply pdata/plog/log_record_flags.go:25]: My IDE got crazy and replace that, will fix

Comment (bogdandrutu) [reply pdata/plog/log_record_flags.go]: Done.

File: pdata/pmetric/generated_metrics.go [hunk 2]
```diff
@@ -1846,9 +1856,14 @@ func (ms ExponentialHistogramDataPoint) Exemplars() ExemplarSlice {
 	return ExemplarSlice(internal.NewExemplarSlice(&ms.getOrig().Exemplars))
 }
 
-// Flags returns the flags associated with this ExponentialHistogramDataPoint.
-func (ms ExponentialHistogramDataPoint) Flags() MetricDataPointFlags {
```
Comment (dmitryax) [inline pdata/pmetric/generated_metrics.go:1850]: Don't we need to keep the method as deprecated?

Comment (dmitryax) [reply pdata/pmetric/generated_metrics.go:1850]: nvm. found it

File: CHANGELOG.md [hunk 3]
```diff
@@ -36,6 +36,13 @@
 - Enforce naming conventions for Invalid[Trace|Span]ID: (#5969)
   - Deprecate funcs `pcommon.InvalidTraceID` and `pcommon.InvalidSpanID` in favor of vars `pcommon.EmptyTraceID` and `pcommon.EmptySpanID` 
 - Deprecate `Update` and `Upsert` methods of `pcommon.Map` (#5975)
+- Deprecated the current MetricDataPointFlags API.  The new API provides functions to check and set Flags. (#5999)
+  - `NumberDataPoint.Flags` -> `NumberDataPoint.FlagsImmutable`
+  - `HistogramDataPoint.Flags` -> `HistogramDataPoint.FlagsImmutable`
+  - `ExponentialHistogramDataPoint.Flags` -> `ExponentialHistogramDataPoint.FlagsImmutable`
+  - `SummaryDataPoint.Flags` -> `SummaryDataPoint.FlagsImmutable`
+  - `MetricDataPointFlags` -> `MetricDataPointFlagsImmutable`
+  - `NewMetricDataPointFlags` -> `MetricDataPointFlagsImmutable`
```
Comment (dmitryax) [inline CHANGELOG.md:45]: Will we rename it back? We probably should mention that in the comments and here?

Comment (bogdandrutu) [reply CHANGELOG.md:45]: Already done https://github.com/open-telemetry/opentelemetry-collector/pull/5999/files#diff-39b6dfa0a0fdf8e13fb63075018e16a009a905cfd91c451af47768bac6268e96R24

**Review by dmitryax — APPROVED** [review]

**Pull request merged**